pkgs/development/python-modules/scalene/default.nix

```diff
@@ -2,33 +2,65 @@
   lib,
   stdenv,
   buildPythonPackage,
-  hypothesis,
-  fetchPypi,
-  setuptools,
-  setuptools-scm,
+  fetchFromGitHub,
   cloudpickle,
   cython,
+  hypothesis,
   jinja2,
   numpy,
+  nvidia-ml-py,
   psutil,
   pynvml,
-  nvidia-ml-py,
   pytestCheckHook,
   pythonOlder,
   rich,
+  setuptools-scm,
+  setuptools,
 }:
 
 buildPythonPackage rec {
   pname = "scalene";
-  version = "1.5.43.2";
+  version = "1.5.44.1";
   pyproject = true;
   disabled = pythonOlder "3.9";
 
-  src = fetchPypi {
-    inherit pname version;
-    hash = "sha256-LtD7v3pLz4UCnh6xlhkPdcEjyu3mt+YQPYZ0nNCLuDw=";
+  src = fetchFromGitHub {
+    owner = "plasma-umass";
+    repo = "scalene";
+    rev = "v${version}";
+    hash = "sha256-XMz+gwiNaKiKplD4kOE1yhcg+dkzjEdDYjW0JsDEMQE=";
+  };
+
+  heap-layers-src = fetchFromGitHub {
+    owner = "emeryberger";
+    repo = "heap-layers";
+    name = "Heap-Layers";
+    rev = "a2048eae91b531dc5d72be7a194e0b333c06bd4c";
+    sha256 = "sha256-vl3z30CBX7hav/DM/UE0EQ9lLxZF48tMJrYMXuSulyA=";
   };
 
+  printf-src = fetchFromGitHub {
+    owner = "mpaland";
+    repo = "printf";
+    name = "printf";
+    rev = "v4.0.0";
+    sha256 = "sha256-tgLJNJw/dJGQMwCmfkWNBvHB76xZVyyfVVplq7aSJnI=";
+  };
```

Comment:
`heap-layers-src` and `printf-src` don't need to be parameters to `buildPythonPackage`. It would be preferable to put them in a `let ... in` block above `buildPythonPackage`.

Reply:
Doh, of course! Thank you.